crates/broker/src/order_picker.rs

```diff
@@ -462,47 +503,93 @@ where
+
+            // 5 second interval with a 2.5 second delay so config is checked first,
+            // and that the requests are interleaved between config checks.
+            let pricing_check_interval = tokio::time::Duration::from_secs(5);
+            let mut pricing_check_timer = tokio::time::interval_at(
+                tokio::time::Instant::now() + tokio::time::Duration::from_millis(2500),
+                pricing_check_interval,
+            );
+
             loop {
-                let order_res = picker_copy
-                    .db
-                    .get_order_for_pricing()
-                    .await
-                    .map_err(|err| SupervisorErr::Recover(err.into()))?;
-
-                if let Some((order_id, order)) = order_res {
-                    let picker_clone = picker_copy.clone();
-                    // TODO: We should consider having handles for these inner tasks
-                    // but they are one-shots that self-clean up on the DB so maybe its fine?
-                    tokio::spawn(async move {
-                        if let Err(err) = picker_clone.price_order(order_id, &order).await {
-                            picker_clone
-                                .db
-                                .set_order_failure(order_id, err.to_string())
-                                .await
-                                .expect("Failed to set order failure");
-                            match err {
-                                PriceOrderErr::OtherErr(err) => {
-                                    tracing::error!("Pricing order failed: {order_id:x} {err:?}");
+                tokio::select! {
+                    _ = config_check_timer.tick() => {
+                        // Get updated max concurrent locks and calculate capacity based on orders
+                        // that are locked but not fulfilled yet.
+                        capacity = picker_copy
+                            .get_pricing_order_capacity()
+                            .await
+                            .map_err(SupervisorErr::Recover)?;
+                    }
+
+                    _ = pricing_check_timer.tick() => {
+                        // Queue up orders that can be added to capacity.
+                        let order_size = if let Some(capacity) = capacity {
+                            capacity.saturating_sub(
+                                u32::try_from(pricing_tasks.len()).expect("tasks u32 overflow"),
+                            )
+                        } else {
+                            // If no maximum lock capacity, request a max of 10 orders at a time.
+                            10
+                        };
+
+                        picker_copy
+                            .spawn_pricing_tasks(&mut pricing_tasks, order_size)
+                            .await
+                            .map_err(SupervisorErr::Recover)?;
```

Comment:
If `order_size` is zero, this seems to avoid kicking off any pricing tasks, but it does still query the DB.

Reply:
There is the `capacity == 0` check in that method. I can move it to wrap this call, if preferred?